packages-next/keystone/src/lib/context/createFilesContext.ts

```diff
@@ -0,0 +1,63 @@
+import path from 'path';
+import { v4 as uuid } from 'uuid';
+import filenamify from 'filenamify';
+import { FilesConfig, FilesContext } from '@keystone-next/types';
+import fs from 'fs-extra';
+
+import { parseFileRef } from '@keystone-next/utils-legacy';
+
+const DEFAULT_BASE_URL = '/files';
+const DEFAULT_STORAGE_PATH = './public/files';
+
+const generateSafeFilename = (filename: string) => {
+  //   /*
+  //     This regex lazily matches for any characters that aren't a new line
+  //     it then optionally matches the last instance of a "." symbol
+  //     followed by any alphabetical character before the end of the string
+  //    */
+  const [, name, ext] = filename.match(/^([^:\n].*?)(\.[A-Za-z]+)?$/) as RegExpMatchArray;
+
+  const id = uuid();
+  const urlSafeName = filenamify(name, { replacement: '_' }).toLowerCase();
+  if (ext) {
+    return `${urlSafeName}_${id}${ext}`;
+  }
+  return `${urlSafeName}_${id}`;
+};
+
+export function createFilesContext(config?: FilesConfig): FilesContext | undefined {
+  if (!config) {
+    return;
+  }
+
+  const { baseUrl = DEFAULT_BASE_URL, storagePath = DEFAULT_STORAGE_PATH } = config.local || {};
+
+  fs.mkdirSync(storagePath, { recursive: true });
+
+  return {
+    getSrc: (mode, filename) => {
+      return `${baseUrl}/${filename}`;
+    },
+    getDataFromRef: async (ref: string) => {
+      const fileRef = parseFileRef(ref);
+      if (!fileRef) {
+        throw new Error('Invalid file reference');
+      }
+      const { size: filesize } = await fs.stat(path.join(storagePath, `${fileRef.filename}`));
+      return { filesize, ...fileRef };
+    },
+    getDataFromStream: async (stream, filename) => {
+      const { upload: mode } = config;
+
+      const pseudoSafeFilename = generateSafeFilename(filename);
+      const writeStream = fs.createWriteStream(path.join(storagePath, pseudoSafeFilename));
```

Comment:
Can you try uploading a file that is above the max file size and see what happens and then probably handle whatever error that results in?

Reply:
yup, validated this yesterday we get a "payloadtoolarge" error type like so 
<img width="875" alt="payload-too-large" src="https://user-images.githubusercontent.com/12119389/116486603-61f4da80-a8d1-11eb-94f7-0f42e9f80cc4.png">

Of course it's a nested error, so its obfuscated in the admin-ui, but that's no better or worse than what we had previously 😂. Ideally though we'd be able to append a better error message though, something like 
`file exceeds max file size specified in config, see config documentation for details`

Reply:
Either way pulling this out of the PR for now, as per review with @timleslie

Reply:
This code needs to delete the file when an error occurs during the reading correctly regardless of how we configure the max file size and correctly handling that should happen in this PR imo.

Reply:
> Do we delete the file once that happens?

No, we'll need to add logic to delete the file if the error occurs